lib/src/facts/windows/operating_system_resolver.cc

```diff
@@ -101,7 +104,26 @@ namespace facter { namespace facts { namespace windows {
         auto version = result.release.substr(0, lastDot);
         bool consumerrel = (wmi::get(vals, wmi::producttype) == "1");
         if (version == "10.0") {
-            result.release = consumerrel ? "10" : "2016";
+            // Calculate the build number to distinguish between
+            // Windows Server 2016 and 2019. Note that the kernel
+            // version is written as <major>.<minor>.<build_number>
+            auto kernel_version_fact = facts.get<string_value>(fact::kernel_version);
+            if (! kernel_version_fact) {
+                LOG_DEBUG("Could not resolve the OS release and OS major version facts from the kernel version fact");
+                return result;
```

Comment:
Yes because these types of changes query system calls and those are tough to mock out in C++. For example, there are no unit tests for the Windows operating system resolver.

Reply:
This code is covered by the acceptance tests.

Reply:
Coolness.

Reply:
@ekinanp Is that to say the acceptance tests are failing until this goes in?

Reply:
No, the acceptance tests are passing for our suite pipelines. However once we add Server 2019 to agent CI, then the acceptance tests will fail without this change.